doc/modules/linear_model.rst

```diff
@@ -126,6 +126,25 @@ its ``coef_`` member::
     >>> reg.intercept_
     0.13636...
 
+Note that the class :class:`Ridge` allows for the user to specify that the 
+solver be automatically chosen when ``solver=auto``. When this option 
+is specified, :class:`Ridge` will choose between the "lbfgs", "sag", 
+"cholesky", and "sparse_cg" solvers. :class:`Ridge` will begin checking 
+the conditions shown in the following table from top to bottom. If the 
+condition is true, the corresponding solver is chosen.
+
++-------------+----------------------------------------------------+
+| **Solver**  | **Condition**                                      |
++-------------+----------------------------------------------------+
+| 'lbfgs'     | The ``positive=True`` option is specified.         |
++-------------+----------------------------------------------------+
+| 'sag'       | The ``return_intercept=True`` option is specified. |
```

Comment:
`return_intercept` is an artifact of the private API that does not directly map to any parameter of the public API. Here it would imply that `"sag"` is used by default by `Ridge` for dense data while actually `"cholesky"` is used (I believe).

I would just drop this line from the table. Alternatively one could use a sentence such as:

> When `solver="auto"`, :class:`Ridge` uses the `"cholesky"` solver for dense data and the `"sparse_cg"` solver for sparse data, unless `positive=True` is set, in which case the `"lbfgs"` solver will be used for either dense or sparse data.

I believe a similar statement should be added to the docstring of the `Ridge` and `RidgeCV` classes.

Reply:
Ah, I understand. I can strike the `"sag"` line from the table. I do really like your one-sentence description though. Do you think it would be okay to have both the table (without `"sag"`) and your description?